protocol/test/foundry/PipelineConvert.t.sol

```diff
@@ -0,0 +1,1461 @@
+        int96[] memory stems = new int96[](1);
+        stems[0] = stem;
+
+        AdvancedFarmCall[] memory farmCalls = new AdvancedFarmCall[](1);
+        AdvancedFarmCall[] memory beanToLPFarmCalls = createLPToBeanFarmCalls(lpAmountOut);
+        farmCalls[0] = beanToLPFarmCalls[0]; // Assign the first element of the returned array
+
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = lpAmountOut;
+        
+        vm.resumeGasMetering();
+        vm.prank(users[1]); // do this as user 1
+        convert.pipelineConvert(
+            C.BEAN_ETH_WELL, // input token
+            stems,  // stems
+            amounts,  // amount
+            C.BEAN, // token out
+            farmCalls // farmData
+        );
+    }
+
+
+
+    function testBasicConvertLPToLP(uint256 amount) public {
+        vm.pauseGasMetering();
+
+        // well is initalized with 10000 beans. cap add liquidity 
+        // to reasonable amounts.
+        amount = bound(amount, 1e6, 5000e6);
+
+        (int96 stem, uint256 lpAmountOut) = depositLPAndPassGermination(amount);
+
+        mineBlockAndUpdatePumps();
+
+        // log deltaB for this well before convert
+        int256 beforeDeltaBEth = seasonGetters.poolDeltaBInsta(C.BEAN_ETH_WELL);
+        int256 beforeDeltaBwsteth = seasonGetters.poolDeltaBInsta(C.BEAN_WSTETH_WELL);
+
+        // uint256 beforeGrownStalk = bs.balanceOfGrownStalk(users[1], C.BEAN_ETH_WELL);
+        uint256 beforeBalanceOfStalk = bs.balanceOfStalk(users[1]);
+
+        int96[] memory stems = new int96[](1);
+        stems[0] = stem;
+
+        AdvancedFarmCall[] memory farmCalls = new AdvancedFarmCall[](1);
+        AdvancedFarmCall[] memory lpToLPFarmCalls = createLPToLPFarmCalls(lpAmountOut);
+        farmCalls[0] = lpToLPFarmCalls[0]; // Assign the first element of the returned array
+
+
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = lpAmountOut;
+        
+        vm.resumeGasMetering();
+        vm.prank(users[1]);
+
+        convert.pipelineConvert(
+            C.BEAN_ETH_WELL, // input token
+            stems,  // stems
+            amounts,  // amount
+            C.BEAN_WSTETH_WELL, // token out
+            farmCalls // farmData
+        );
+    
+        int256 afterDeltaBEth = seasonGetters.poolDeltaBInsta(C.BEAN_ETH_WELL);
+        int256 afterDeltaBwsteth = seasonGetters.poolDeltaBInsta(C.BEAN_WSTETH_WELL);
+
+        // make sure deltaB's moved in the way we expect them to
+        assertTrue(beforeDeltaBEth < afterDeltaBEth);
+        assertTrue(afterDeltaBwsteth < beforeDeltaBwsteth);
+
+        uint256 totalStalkAfter = bs.balanceOfStalk(users[1]);
+
+        // since we didn't cross peg and there was convert power, we expect full remaining grown stalk
+        // (plus a little from the convert benefit)
+        assertTrue(totalStalkAfter >= beforeBalanceOfStalk);
+    }
+
+
+    function testUpdatingOverallDeltaB(uint256 amount) public {
```

Comment:
this test can be improved by:
1) randomizing which wells bean are added to (beanEth, beanwstETH, or both)
2) directly calculating what overall deltaB should be and using that as an assertion.

Reply:
Improved to do both of those here: https://github.com/BeanstalkFarms/Beanstalk/pull/840/commits/741220c21640e5323e485590c225f8147e409314